File: frontend/src/types/project.ts
export type ProjectStatus = "active" | "completed" | "on-hold";

export interface Project {
  id: string;
  title: string;
  description: string;
  status: ProjectStatus;
  client_id?: string | null;
  client_name?: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateProjectData {
  title: string;
  description: string;
  status: ProjectStatus;
  client_id?: string | null;
}

export interface UpdateProjectData {
  title?: string;
  description?: string;
  status?: ProjectStatus;
  client_id?: string | null;
}

File: backend/app/domain/entities/project.py
from datetime import datetime
from typing import Optional
from app.domain.value_objects.project_status import ProjectStatus


class Project:
    def __init__(
        self,
        id: Optional[str] = None,
        title: str = "",
        description: str = "",
        status: ProjectStatus = ProjectStatus.ACTIVE,
        client_id: Optional[str] = None,
        created_at: Optional[datetime] = None,
        updated_at: Optional[datetime] = None,
    ):
        self.id = id
        self.title = title
        self.description = description
        self.status = status
        self.client_id = client_id
        self.created_at = created_at or datetime.utcnow()
        self.updated_at = updated_at or datetime.utcnow()

